x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigInitialNode.java

```diff
@@ -0,0 +1,404 @@
+                fullyWriteFile(transportKeystoreOutput, stream -> transportKeystore.store(stream, transportKeystorePassword.getChars()));
+                nodeKeystore.setString("xpack.security.transport.ssl.keystore.secure_password", transportKeystorePassword.getChars());
+                // we use the same PKCS12 file for the keystore and the truststore
+                nodeKeystore.setString("xpack.security.transport.ssl.truststore.secure_password", transportKeystorePassword.getChars());
+            }
+            Path httpKeystoreOutput = instantAutoConfigDir.resolve(HTTP_AUTOGENERATED_KEYSTORE_NAME + ".p12");
+            try (SecureString httpKeystorePassword = newKeystorePassword()) {
+                KeyStore httpKeystore = KeyStore.getInstance("PKCS12");
+                httpKeystore.load(null);
+                // the keystore contains both the node's and the CA's private keys
+                // both keys are encrypted using the same password as the PKCS12 keystore they're contained in
+                httpKeystore.setKeyEntry(HTTP_AUTOGENERATED_KEYSTORE_NAME + "_ca", httpCAKeyPair.getPrivate(),
+                        httpKeystorePassword.getChars(), new Certificate[]{httpCACert});
+                httpKeystore.setKeyEntry(HTTP_AUTOGENERATED_KEYSTORE_NAME, httpKeyPair.getPrivate(),
+                        httpKeystorePassword.getChars(), new Certificate[]{httpCert, httpCACert});
+                fullyWriteFile(httpKeystoreOutput, stream -> httpKeystore.store(stream, httpKeystorePassword.getChars()));
+                nodeKeystore.setString("xpack.security.http.ssl.keystore.secure_password", httpKeystorePassword.getChars());
+            }
+            // finally overwrites the node keystore (if the keystores have been successfully written)
+            nodeKeystore.save(env.configFile(), nodeKeystorePassword.getChars());
+        }
+
+        // at this point the elasticsearch.keystore is updated but the elasticsearch.yml file not yet
+
+        // TODO atomic write
+        try (BufferedWriter bw = Files.newBufferedWriter(ymlPath, StandardCharsets.UTF_8, StandardOpenOption.APPEND)) {
+            bw.newLine();
+            bw.newLine();
+            bw.write("###################################################################################");
+            bw.newLine();
+            bw.write("# The following settings, and associated TLS certificates and keys configuration, #");
+            bw.newLine();
+            bw.write("# have been automatically generated in order to configure Security.               #");
+            bw.newLine();
+            bw.write("# These have been generated the first time that the new node was started, without #");
+            bw.newLine();
+            bw.write("# joining or enrolling to an existing cluster and only if Security had not been   #");
+            bw.newLine();
+            bw.write("# explicitly configured beforehand.                                               #");
```

Comment:
Good place to describe what to do if it wasn't intended to be a new cluster?

Reply:
Can you please add this comment to the review of the PR that introduces this code and not here?